core/tests/upgrade-test/tests/utils.ts

```diff
@@ -0,0 +1,125 @@
+import { ethers } from 'ethers';
+import * as fs from 'fs';
+import { background } from 'utils';
+import { getConfigPath } from 'utils/build/file-configs';
+
+export function runServerInBackground({
+    components,
+    stdio,
+    cwd,
+    useZkInception
+}: {
+    components?: string[];
+    stdio: any;
+    cwd?: Parameters<typeof background>[0]['cwd'];
+    useZkInception?: boolean;
+}) {
+    let command = useZkInception
+        ? 'zk_inception server'
+        : 'cd $ZKSYNC_HOME && cargo run --bin zksync_server --release --';
+    if (components && components.length > 0) {
+        command += ` --components=${components.join(',')}`;
+    }
+    background({ command, stdio, cwd });
+}
+
+export function setEthSenderSenderAggregatedBlockCommitDeadline(pathToHome: string, fileConfig: any, value: number) {
+    setPropertyInGeneralConfig(pathToHome, fileConfig, 'aggregated_block_commit_deadline', value);
+}
+
+export function setAggregatedBlockProveDeadline(pathToHome: string, fileConfig: any, value: number) {
+    setPropertyInGeneralConfig(pathToHome, fileConfig, 'aggregated_block_prove_deadline', value);
+}
+
+export function setAggregatedBlockExecuteDeadline(pathToHome: string, fileConfig: any, value: number) {
+    setPropertyInGeneralConfig(pathToHome, fileConfig, 'aggregated_block_execute_deadline', value);
+}
+
+export function setBlockCommitDeadlineMs(pathToHome: string, fileConfig: any, value: number) {
+    setPropertyInGeneralConfig(pathToHome, fileConfig, 'block_commit_deadline_ms', value);
+}
+
+function setPropertyInGeneralConfig(pathToHome: string, fileConfig: any, property: string, value: number) {
+    const generalConfigPath = getConfigPath({
+        pathToHome,
+        chain: fileConfig.chain,
+        configsFolder: 'configs',
+        config: 'general.yaml'
+    });
+    const generalConfig = fs.readFileSync(generalConfigPath, 'utf8');
+    const regex = new RegExp(`\\b${property}:\\s*\\d+`, 'g');
+    const newGeneralConfig = generalConfig.replace(regex, `${property}: ${value}`);
+
+    fs.writeFileSync(generalConfigPath, newGeneralConfig, 'utf8');
+}
+
+export interface Contracts {
+    l1DefaultUpgradeAbi: any;
+    governanceAbi: any;
+    adminFacetAbi: any;
+    chainAdminAbi: any;
+    l2ForceDeployUpgraderAbi: any;
+    complexUpgraderAbi: any;
+    counterBytecode: any;
+    stateTransitonManager: any;
+}
+
+export function initContracts(zkToolbox: boolean): Contracts {
+    if (zkToolbox) {
+        const CONTRACTS_FOLDER = `${process.env.ZKSYNC_HOME}/contracts`;
```

Comment:
don't use ZKSYNC_HOME

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/2515/commits/0f31c9ff8b170ebd32aa9fe32f99458a46610628